match_p.h

```diff
@@ -33,10 +33,8 @@
 
 
 #if defined(_MSC_VER) && !defined(__clang__)
-# if defined(_M_IX86) || defined(_M_AMD64) || defined(_M_IA64)
+# if defined(_M_IX86) || defined(_M_AMD64) || defined(_M_IA64) ||  defined(_M_ARM) || defined(_M_ARM64)
 #  include "arch/x86/ctzl.h"
```

Comment:
What do you mean?

Reply:
We need to move the header out of arch/x86 and possibly rename it to make it clear it is used for multiple archs using Visual C++ compiler.

Reply:
Hmm. Are you suggesting to put it in the arch/generic folder?

Reply:
Or I could rename it to arch/x86/msvc_ctzl.h

Reply:
We haven't really decided how we are going to use arch/generic as most of the generic *_c functions are still in the project root directory.

Reply:
Yeah I don't really have a good idea here.

Reply:
Maybe /msvc_ctzl.h until we start to populate arch/generic...

Reply:
@Dead2 thoughts?

Reply:
I am thinking that this is not really arch-specific, and it does not really fit into generic either.
The thought behind generic was to have for example fill_window_c there, and that would be used if there was no fill_window_X implementation in the activated arch. So sort of a place for the fallback code that has arch-specific versions. It would also simplify keeping it and the arch-specific versions in sync properly, both through better visibility and by easier diffing.

In this case, I suggest moving it into /fallback-builtins.h as what it is is a fallback implementation of a compiler builtin. And we might end up getting more of those at some point. We might also need arch-specific builtins at some point, those should go into the arch folder.

Reply:
Done